pkg/config/config.go

```diff
@@ -6,18 +6,38 @@
 package config
 
 import (
+	"fmt"
 	"os"
 	"strings"
 
-	"github.com/go-logr/logr"
+	"golang.org/x/exp/maps"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/labels"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+
+	"github.com/DataDog/datadog-operator/apis/datadoghq/common"
+	datadoghqv1alpha1 "github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	datadoghqv2alpha1 "github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
 )
 
 const (
-	// WatchNamespaceEnvVar is the constant for env variable WATCH_NAMESPACE
-	// which specifies the Namespace to watch.
+	// AgentWatchNamespaceEnvVar is a comma separated list of namespaces watched by Agent controller.
+	agentWatchNamespaceEnvVar = "AGENT_WATCH_NAMESPACE"
+	// SLOWatchNamespaceEnvVar is a comma separated list of namespaces SLO controller.
+	sloWatchNamespaceEnvVar = "SLO_WATCH_NAMESPACE"
+	// MonitorWatchNamespaceEnvVar is a comma separated list of namespaces Monitor controller.
+	monitorWatchNamespaceEnvVar = "MONITOR_WATCH_NAMESPACE"
+	// ProfilesWatchNamespaceEnvVar is a comma separated list of namespaces Agent Profile controller.
+	profileWatchNamespaceEnvVar = "PROFILE_WATCH_NAMESPACE"
+	// WatchNamespaceEnvVar is a comma separated list of namespaces watched by all controllers unless controller specific configuration is provided.
```

Comment:
what do you think about prefixing with `DD_` consistent with how we define other envvars?

```suggestion
	// AgentWatchNamespaceEnvVar is a comma-separated list of namespaces watched by the DatadogAgent controller.
	agentWatchNamespaceEnvVar = "DD_AGENT_WATCH_NAMESPACE"
	// SLOWatchNamespaceEnvVar is a comma-separated list of namespaces watched by the DatadogSLO controller.
	sloWatchNamespaceEnvVar = "DD_SLO_WATCH_NAMESPACE"
	// MonitorWatchNamespaceEnvVar is a comma-separated list of namespaces watched by the DatadogMonitor controller.
	monitorWatchNamespaceEnvVar = "DD_MONITOR_WATCH_NAMESPACE"
	// ProfilesWatchNamespaceEnvVar is a comma-separated list of namespaces watched by the DatadogAgentProfile controller.
	profileWatchNamespaceEnvVar = "DD_AGENT_PROFILE_WATCH_NAMESPACE"
	// WatchNamespaceEnvVar is a comma-separated list of namespaces watched by all controllers, unless a controller-specific configuration is provided.
```

Reply:
I just followed the pattern of existing `WATCH_NAMESPACE` env var. Using DD makes sense with the new ones.